File: EasyInventoryAPI/src/easyinventoryapi/StorageCapacity.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package easyinventoryapi;

import org.json.JSONException;
import org.json.JSONObject;

/**
 *
 * @author 3041TAN-06
 */
public class StorageCapacity {
    
    public final Item item;
    public final int stored_amount;
    public final int limit;
    
    public StorageCapacity(JSONObject o) throws JSONException {
        this.item=new Item(o.getJSONObject("item"));
        this.stored_amount=o.getInt("stored_amount");
        this.limit=o.getInt("limit");
    }
    
}

File: api/src/Logic/StorageCapacity.php
<?php

namespace EI\Logic;

class StorageCapacity {
    public readonly Item $item;
    public readonly int $stored_amount;
    public readonly int $limit;

    public function __construct(Item $item, int $stored_amount, int $limit) {
        $this->item = $item;
        $this->stored_amount = $stored_amount;
        $this->limit = $limit;
    }
}